src/main/java/com/google/flourbot/datastorage/FirebaseDataStorage.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.flourbot.datastorage;
+
+import com.google.api.core.ApiFuture;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.DocumentReference;
+
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOptions;
+
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Query;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.QuerySnapshot;
+import com.google.cloud.firestore.WriteResult;
+import com.google.common.collect.ImmutableMap;
+
+import java.io.*;
+
+public class FirebaseDataStorage implements DataStorage {
+    private Firestore db;
+
+    public FirebaseDataStorage () {
+        try {
+            this.db = initializeFirebase();
+        }
+        catch (FileNotFoundException e) {
+            System.out.println("Service key not found.");
+        }
+        catch (Exception e) {
+            System.out.println("Firebase not initialized.");
+        }
```

Comment:
Is there a better way for me to handle this exception? For instance, should I be doing System.exit instead of println()?

Reply:
My opinion is these exceptions could be propagated with `FirebaseDataStorage () throws ...` so the servlet that uses `FirebaseDataStorage () ` could return an error message to the hangout chat. I am not sure though, you should check with Marek or Vector.

Reply:
Good question! Not sure either

Reply:
Ok thanks, I will likely use your suggestion Tony and just propagate the error to the server by throwing an exception from the constructor. @Vector-S @lipczak-g is this an acceptable strategy?